views/pages/AdminPage.jsx

```diff
@@ -1,696 +1,201 @@
-      console.error(err.message)
+      const resetKey = crypto.randomUUID()
+      const userId = getUserValues(rowIndex).uid
+      const userAttributes = { force_reset_pw: true, reset_key: resetKey }
+      const updatedUser = await api.admin.users.update(userId, userAttributes)
+
+      updateUserInForm(rowIndex, updatedUser)
+      setResetKeyValue(resetKey)
+      setOpenResetKeyModal(true)
+      handleNotification('Reset Key has been applied successfully')
+    } catch (error) {
+      handleNotification(error.message)
     }
   }
-  componentWillUnmount() {
-    window.removeEventListener('resize', this.handleResize)
-  }
-  handleDrawerToggle = () => {
-    this.setState((state) => ({ mobileOpen: !state.mobileOpen }))
+  const handleUserRole = (rowIndex) => {
+    setOpenAccessModal(true)
+    setCurrentRowIndex(rowIndex)
   }
-  autocomplete = ({ users }) =>
-    users.map((option) => ({
-      value: option.uid,
-      label:
-        /\S/.test(option.display_name) && option.display_name != undefined
-          ? option.display_name
-          : option.uid,
-    }))
-  handleTab = (value) => {
-    this.setState({
-      tab: value,
-    })
+  const handleClose = () => {
+    setResetKeyValue('')
+    setOpenResetKeyModal(false)
+    setOpenAccessModal(false)
+    setCurrentRowIndex(null)
   }
-  handleResize = () => {
-    this.setState({
-      width: window.innerWidth - this.state.marginWidth,
-      height: window.innerHeight - this.state.marginHeight,
-    })
+  const handleUpdateUserAccess = async (rowIndex) => {
+    try {
+      const userValues = getUserValues(rowIndex)
+      const userId = userValues.uid
+      const userAttributes = UsersModel.createUserFromFormValues(userValues)
+      const updatedUser = await api.admin.users.update(userId, userAttributes)
+      const userToFormValues = UsersModel.createFormValuesFromUser(updatedUser)
+
+      updateUserInForm(rowIndex, userToFormValues)
+      handleClose()
+      handleNotification('User updated successfully')
+    } catch (error) {
+      handleNotification(error.message)
+    }
   }
-  sort = ({ sortBy, sortDirection }) => {
-    const sortedList = this.sortList({ sortBy, sortDirection })
-    this.setState({
-      sortBy: sortBy,
-      sortDirection: sortDirection,
-      users: sortedList,
-    })
+  const handleUserUpdate = async (data, userId) => {
+    try {
+      const user = data.users.find((user) => user.uid === userId)
+      const userAttributes = UsersModel.createUserFromFormValues(user)
+
+      await api.admin.users.update(userId, userAttributes)
+
+      handleNotification('User updated successfully')
+    } catch (error) {
+      handleNotification(error.message)
+    }
   }
-  sortList = ({ sortBy, sortDirection }) => {
-    let list = _.map(this.state.users, _.clone)
-    return _.orderBy(list, [sortBy], sortDirection.toLowerCase())
+  const fetchStudies = async () => {
+    try {
+      const listOfStudies = await api.admin.studies.all()
```

Comment:
should this use the studies that a user has access to or should it be every one?